### PR TITLE
Match exactly tweaks and polish

### PR DESCRIPTION
**Pull request by seadowg**
~~Blocked by #4020~~ 

Changes:

* Use [Material "Sync problem" icon](https://material.io/resources/icons/?icon=sync_problem&style=baseline) for refresh button when there has been an error
* Correct size of other icons in app bar on form list pages
* Use "Manual" instead of "Manually" for form mode
* Show a toast when the match exactly sync is successful (and was kicked off manually)
* Show progress bar for a minimum amount of time so on form list pages so it's obvious to the user things are happening

#### What has been done to verify that this works as intended?

New tests and verified manually.

#### Why is this the best possible solution? Were any other approaches considered?

I flipped flopped on showing the toast in the `Notifier` implementation but eventually decided it was just simpler to have that be a concern of the menu delegate.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Good to check manual updates for Match Exactly as that's where the most has changed.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/res/drawable/ic_baseline_refresh_error_24.xml [hunk 1]
```diff
@@ -3,8 +3,8 @@
     android:height="24dp"
     android:viewportWidth="24"
     android:viewportHeight="24"
-    android:tint="?attr/colorError">
+    android:tint="?attr/colorControlNormal">
   <path
       android:fillColor="@android:color/white"
-      android:pathData="M17.65,6.35C16.2,4.9 14.21,4 12,4c-4.42,0 -7.99,3.58 -7.99,8s3.57,8 7.99,8c3.73,0 6.84,-2.55 7.73,-6h-2.08c-0.82,2.33 -3.04,4 -5.65,4 -3.31,0 -6,-2.69 -6,-6s2.69,-6 6,-6c1.66,0 3.14,0.69 4.22,1.78L13,11h7V4l-2.35,2.35z"/>
+      android:pathData="M3,12c0,2.21 0.91,4.2 2.36,5.64L3,20h6v-6l-2.24,2.24C5.68,15.15 5,13.66 5,12c0,-2.61 1.67,-4.83 4,-5.65L9,4.26C5.55,5.15 3,8.27 3,12zM11,17h2v-2h-2v2zM21,4h-6v6l2.24,-2.24C18.32,8.85 19,10.34 19,12c0,2.61 -1.67,4.83 -4,5.65v2.09c3.45,-0.89 6,-4.01 6,-7.74 0,-2.21 -0.91,-4.2 -2.36,-5.64L21,4zM11,13h2L13,7h-2v6z"/>
```
Comment (seadowg) [inline collect_app/src/main/res/drawable/ic_baseline_refresh_error_24.xml:9]: This icon is intended for representing sync problems in Material. Perfect!

File: async/src/main/java/org/odk/collect/async/CoroutineAndWorkManagerScheduler.kt [hunk 2]
```diff
@@ -7,13 +7,20 @@ import java.util.function.Consumer
...
     override fun <T> immediate(background: Supplier<T>, foreground: Consumer<T>) {
-        CoroutineScope(this.foreground).launch {
-            foreground.accept(withContext(this@CoroutineAndWorkManagerScheduler.background) { background.get() })
+        CoroutineScope(foregroundContext).launch {
+            val result = withContext(backgroundContext) { background.get() }
+            foreground.accept(result)
+        }
+    }
+
+    override fun immediate(foreground: java.lang.Runnable) {
```
Comment (seadowg) [inline async/src/main/java/org/odk/collect/async/CoroutineAndWorkManagerScheduler.kt:21]: This ended up not being needed for anything but will be nice to have in the future I think.

File: collect_app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -695,7 +695,7 @@
     <string name="legacy_custom_server_paths_summary">Will be removed in a future version. Please use /formList and /submission on your server.</string>
     <string name="experimental">Experimental</string>
     <string name="match_exactly">Exactly match server</string>
-    <string name="manually">Manually</string>
+    <string name="manually">Manual</string>
```
Comment (lognaturel) [inline collect_app/src/main/res/values/strings.xml]: While here, can I ask you to change the resource name. 🙈

File: collect_app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -695,11 +695,12 @@
...
     <string name="form_update_mode_title">Blank form update mode</string>
     <string name="previously_downloaded_only">Previously downloaded forms only</string>
     <string name="form_update_frequency_title">Automatic update frequency</string>
     <string name="form_update_error">Form update failed</string>
     <string name="report_to_project_lead">If you keep having this problem, report it to the person who asked you to collect data.</string>
     <string name="unknown_host_error">Collect can\'t reach the server at %s. Did you enter the URL correctly?</string>
+    <string name="form_update_complete">Form update complete</string>
```
Comment (lognaturel) [inline collect_app/src/main/res/values/strings.xml]: How about "Form update succeeded" or "Form update completed successfully"?

**Review by lognaturel — CHANGES_REQUESTED** [review]
Great! Currently the form update icon is still visible when searching. Not a deal breaker but probably shouldn't be maintained.

**Comment by seadowg** [comment]
> Currently the form update icon is still visible when searching. Not a deal breaker but probably shouldn't be maintained.

This is a little awkward to fix but I'll see if I can squeeze it into this PR!

**Comment by seadowg** [comment]
@lognaturel ok fixed! Ran into some nasty Robolectric/Support library problems with testing but think I might have a slightly less Robolectric intense process for testing menu delegates now.

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Just needs a rebase on master and then can be tagged for testing. 👍

**Review by lognaturel — CHANGES_REQUESTED** [review]
I just realized that this changes all progress bars. I don't think that's right. E.g. there's a progress bar shown each time a form list loads. Generally this should be instant and there's only ever work done if there are new files on disk to be scanned. Now the progress bar makes it feel like something is happening each time.

**Comment by seadowg** [comment]
@lognaturel yeah I agree, it felt right at first but after playing with it more I'm not liking it either.

**Review by lognaturel — APPROVED** [review]

**Comment by mmarciniak90** [comment]
> Show progress bar for a minimum amount of time so on form list pages so it's obvious to the user things are happening

@seadowg could you give me some clue about this? To be honest, I can't notice a difference.

Verified cases:
* refresh button when there has been an error
* size of other icons in app bar on all pages
* "Manual" instead of "Manually" 
*  toast when the match exactly sync is successful 
* light/dark mode
* RTL language

|Screenshot_20200902-154551|Screenshot_20200902-154611|Screenshot_20200902-154625|Screenshot_20200902-154519|
|---|---|---|---|

**Comment by seadowg** [comment]
@mmarciniak90 just that when you hit refresh (🔃) in "Fill Blank Form" with match exactly enabled the progress bar should show for a half second even if the network request is faster than that. Nothing to verify there other than that the progress bar doesn't seem to be broken in any way!

**Comment by mmarciniak90** [comment]
>  the progress bar should show for a half second

@seadowg hmm, I can see this progress bar for even 10 seconds when I click on refresh icon when all forms are in device. 

> progress bar doesn't seem to be broken in any way!

Agree

**Comment by seadowg** [comment]
> hmm, I can see this progress bar for even 10 seconds when I click on refresh icon when all forms are in device.

That's fine (I'm assuming the server has a lot of forms or is slow). The problem was that if the server was very fast it looked like nothing had happened.

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Androids: 5.1, 7.1 and 10.

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Pull request merged**